safe_transaction_service/history/indexers/safe_events_indexer.py

```diff
@@ -659,18 +869,53 @@ def _process_decoded_elements(self, decoded_elements: list[EventData]) -> list[A
         )
         if safe_addresses_creation_events:
             # Process safe creation events
+            # Note: When conditional indexing is enabled, events are already filtered
+            # by tx._from in process_elements() before reaching this point
             creation_events_processed = self._process_safe_creation_events(
                 safe_addresses_creation_events
             )
             processed_elements.extend(creation_events_processed)
 
+        elements_to_process = [
+            element
+            for element in decoded_elements
+            if element["event"] not in ("SafeSetup", "ProxyCreation")
+        ]
+
+        # When conditional indexing is enabled, only process events for Safes
+        # that exist in SafeContract table
+        if self.conditional_indexing_enabled:
+            # Get all unique Safe addresses from non-creation events
+            non_creation_addresses = {
+                element["address"] for element in decoded_elements
+            }
+            if non_creation_addresses:
+                # Check which addresses exist in SafeContract
+                existing_addresses = SafeContract.objects.get_existing_addresses(
+                    non_creation_addresses
+                )
+                skipped_count = len(non_creation_addresses) - len(existing_addresses)
+                if skipped_count:
+                    logger.debug(
+                        "Conditional indexing: skipping %d non-creation events "
+                        "for %d Safes not in SafeContract",
+                        skipped_count,
+                        skipped_count,
+                    )
+                # Filter elements to only those with existing SafeContract
+                elements_to_process = [
+                    element
+                    for element in decoded_elements
```

Comment:
Here you want to filter again from decoded_elements or from previous filtered elements_to_process?